data-model/src/main/java/io/micronaut/data/model/jpa/criteria/PersistentEntityCriteriaBuilder.java

```diff
@@ -192,4 +192,13 @@ public interface PersistentEntityCriteriaBuilder extends CriteriaBuilder {
      * @return like predicate
      */
     Predicate regex(Expression<String> x, Expression<String> pattern);
+
+    /**
+     * Checks if array contains given expression. Supported only by Cosmos Db.
+     *
+     * @param x The expression (property)
+     * @param y The expression (value to be contained in the array represented by x property in the db)
+     * @return a new predicate
+     */
+    Predicate arrayContains(Expression<?> x, Expression<?> y);
```

Comment:
Needs since

Reply:
Please make sure it makes some informative error for DBs that doesn't support it.

Reply:
Updated in https://github.com/micronaut-projects/micronaut-data/pull/1813/commits/3d0e36c860c47cf909208ab8f2573f77e38465f6
Basically throw error in AbstractSqlLikeQueryBuilder and added test to verify it works for Azure Cosmos and fails for other dbs.